Makefile

```diff
@@ -1,68 +1,134 @@
+.DEFAULT_TARGET: help
 
-# include .env file and export its env vars
-# (-include to ignore error if it does not exist)
+# Import the .env files and export their values (ignore any error if missing)
 -include .env
+-include .env.dev
 
-# linux: allow shell scripts to be executed
-allow-scripts:; chmod +x ./coverage.sh
+# Set the RPC URL's for each target
+test-fork-testnet: export RPC_URL = $(TESTNET_RPC_URL)
+test-fork-prodnet: export RPC_URL = $(PRODNET_RPC_URL)
+test-fork-holesky: export RPC_URL = "https://holesky.drpc.org"
+test-fork-sepolia: export RPC_URL = "https://sepolia.drpc.org"
 
-# init the repo
-install :; make allow-scripts && forge build
+pre-deploy-testnet: export RPC_URL = $(TESTNET_RPC_URL)
+pre-deploy-prodnet: export RPC_URL = $(PRODNET_RPC_URL)
+deploy-testnet: export RPC_URL = $(TESTNET_RPC_URL)
+deploy-prodnet: export RPC_URL = $(PRODNET_RPC_URL)
 
-# create an HTML coverage report in ./report (requires lcov & genhtml)
-coverage:; ./coverage.sh
-	#
-# run unit tests
-test-unit :; forge test --no-match-path "test/fork/**/*.sol"
+# Set the network ID
+pre-deploy-testnet: export NETWORK = $(TESTNET_NETWORK)
+pre-deploy-prodnet: export NETWORK = $(PRODNET_NETWORK)
+deploy-testnet: export NETWORK = $(TESTNET_NETWORK)
+deploy-prodnet: export NETWORK = $(PRODNET_NETWORK)
+
+# Override the verifier and block explorer parameters (network dependent)
+deploy-testnet: export VERIFIER_TYPE_PARAM = --verifier blockscout
+deploy-testnet: export VERIFIER_URL_PARAM = --verifier-url "https://sepolia.explorer.mode.network/api\?"
+deploy-prodnet: export ETHERSCAN_API_KEY_PARAM = --etherscan-api-key $(ETHERSCAN_API_KEY)
+
+# Set production deployments' flag
+deploy: export DEPLOY_AS_PRODUCTION = true
+
+# Override the fork test mode (existing)
+test-exfork-testnet: export FORK_TEST_MODE = fork-existing
+test-exfork-prodnet: export FORK_TEST_MODE = fork-existing
+test-exfork-holesky: export FORK_TEST_MODE = fork-existing
+test-exfork-sepolia: export FORK_TEST_MODE = fork-existing
+
+TEST_SRC_FILES=$(wildcard test/*.sol test/**/*.sol script/*.sol script/**/*.sol src/escrow/increasing/delegation/*.sol src/libs/ProxyLib.sol)
```

Comment:
I think w/o a comment it's kinda confusing why these particular filenames are here.

Reply:
Does it work with a name like `TEST_COVERAGE_SRC_FILES`?